### ISSUE TITLE
Confusing error when using await in non-async function

### ISSUE THREAD

**Issue by anishkny** [prettier/prettier#9799]
**Prettier 2.2.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEqEE+AFAJT7AA6U++AhgO4M4z4BGDAThQNy0AviAA0ICAAdc0AM7JQPbhCYAFHgjkoGAGxYBPOWI7cGYANZwYAZQmmcUAObIY3dHDFwAthzgATX34AMgyO6AwOcABiENyeDDC4jsggDOgwEKIgABYwntoA6llscDK2YHBWGmw4AG5seslgMoYg9jJw3DAqJg5xyKg67WIAVjIAHgBCJuaWVgyecIH2cP2D7iCjY1b2DtpwAIroEPCr2kMgttzt3MlcPtqZEtz2MPk4vjBZyAAcAAxiTwg7XyJgkySeJQ6NRWYgAjkd4N1JJoUjIALRQOB+PyZbhweE4PHdcJ9JADM7rdqeHDOVyUnZ7Q7HFZktZiGAMDhvD5fJAAJnZJhw2h2AGEIJ5SSASgBWTLodoAFU5mnJ5xqbgAklAArArGBnlIAII6qwwPR7U7tQSCIA)
```sh
--parser babel
```

**Input:**
```jsx
function foo () {
  await bar();
}
```

**Output:**
```jsx
SyntaxError: Unexpected token, expected ";" (2:9)
  1 | function foo () {
> 2 |   await bar();
    |         ^
  3 | }
```

**Expected behavior:**
Honestly, this error message makes no sense.

Perhaps a more relevant message would be:
```
SyntaxError: await is only valid in async function
```

**Comment by thorn0** [comment]
This comment might help with improving this: https://github.com/prettier/prettier/issues/9084#issuecomment-683343109

**Comment by fisker** [comment]
All parsers throw similar errors

`espree` `SyntaxError: Unexpected token bar`
`meriyah` `[1:23]: Unexpected token: 'identifier'`
`flow` ``Unexpected identifier, expected the end of an expression statement (`;`)``

I think there is nothing we can do, unless parsers improve it.

**Comment by thorn0** [comment]
Specifically Babel [can throw "'await' is only allowed within async functions"](https://babeljs.io/en/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABMOdEAoCUiDeAoRRAQwHciYpEAjIgJywG48BfIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=&prettier=false&targets=&version=7.12.9&externalPlugins=), but it somehow trickily depends on its configuration.

**Comment by fisker** [comment]
You are right

```js
require('@babel/parser').parse('function foo () {await bar();}', {sourceType: 'module'})
Uncaught [SyntaxError: 'await' is only allowed within async functions (1:17)
] {
  loc: Position { line: 1, column: 17 },
  pos: 17
}
```

**Comment by fisker** [comment]
```js
require('@babel/parser').parse('function foo () {await bar();}', {sourceType: 'module', errorRecovery: true})
Uncaught [SyntaxError: Unexpected token, expected ";" (1:23)
] {
  loc: Position { line: 1, column: 23 },
  pos: 23
}
```

**Comment by fisker** [comment]
Filed an issue on babel https://github.com/babel/babel/issues/12436

**Comment by sosukesuzuki** [comment]
Fixed by #10345
